holoviews/core/spaces.py

```diff
@@ -495,6 +495,13 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+class OverlayCallable(Callable):
+    """
+    A Callable subclass specifically meant to indicate that the Callable
+    represents an overlay operation between two objects.
+    """
```

Comment:
It doesn't have methods. I'd be okay making it a parameter on a Callable, but that would make it more likely for users use it incorrectly, which is worse than simply not declaring it at all because it has the potential to mess up the other streams on an Overlay.